### PR TITLE
Refactor: Make State/Status/Resource More Readable

### PR DESCRIPTION
**Pull request by daniel-hutao**
## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description

1. Resource -> ResourceStatus in State

```go
type ResourceStatus map[string]interface{}

// State is the single component's state.
type State struct {
	Name           string
	InstanceID     string
	DependsOn      []string
	Options        map[string]interface{}
	ResourceStatus ResourceStatus
}
```

2. State -> ResourceStatusInState, Resource -> ResourceStatusFromRead in Status (outputs) 

`internal/pkg/show/status/output.go`

```go
type Status struct {
	// if ResourceStatusInState == ResourceStatusFromRead,
	// then InlineStatus is set to equals ResourceStatusInState and ResourceStatusFromRead.
	InlineStatus           statemanager.ResourceStatus `yaml:",inline,omitempty"`
	ResourceStatusInState  statemanager.ResourceStatus `yaml:"statusInState,omitempty"`
	ResourceStatusFromRead statemanager.ResourceStatus `yaml:"statusNow,omitempty"`
}
```

3. GetStateOperation -> GetStatusOperation in `plugininstaller`

4. Based on the above changes, all other needed adjustments were adjusted.

## Related Issues

N/A

## New Behavior (screenshots if needed)

Refactoring only

### CODE REVIEW

File: docs/core-concepts/core-concepts.zh.md [hunk 1]
```diff
@@ -54,10 +54,10 @@ _State_ 记录了当下开发运维工具链的状况，包括了每个工具的
 - _State_ 实际上是一个记录了状态的 Map, 定义在[这里](https://github.com/devstream-io/devstream/blob/main/internal/pkg/statemanager/state.go#L24)。
 - Map 中的每一个状态都是一个包含了名称、插件、选项和资源的结构体，定义在[这里](https://github.com/devstream-io/devstream/blob/main/internal/pkg/statemanager/state.go#L16)。
 
-### 4. 资源(Resource)
```
Comment (aFlyBird0) [inline docs/core-concepts/core-concepts.zh.md:57]: 要不直接在 "3. 状态" 里贴一下， State = 工具链元信息（名字,id,depends) + 工具配置 + 资源状态？

感觉目前52行的”_State_ 记录了当下开发运维工具链的状况，包括了每个工具的配置以及当下状态。“比较抽象，如果能稍微图形化一点会更好？

**Review by aFlyBird0 — COMMENTED** [review]
resource -> resourceStatus

ci err:
image

code:
https://github.com/devstream-io/devstream/blob/eb985681271d971c5dbbcfeedcd8f53cd518efea/internal/pkg/statemanager/state_test.go#L84

err:
image

code:
image

File: docs/core-concepts/core-concepts.zh.md (hunk 1 above)
Comment (daniel-hutao) [reply docs/core-concepts/core-concepts.zh.md:57]: 我马上看下

Comment (daniel-hutao) [reply docs/core-concepts/core-concepts.zh.md:57]: https://github.com/devstream-io/devstream/pull/1181/commits/84e102180f0b3d2bb5d5d43b3271191c9f548e0e

**Review by aFlyBird0 — APPROVED** [review]

**Comment by daniel-hutao** [comment]
The "link checker" error isn't raised by this pr.

**Pull request merged**